src/components/widgets/MatchHistory.tsx

```diff
@@ -1,20 +1,38 @@
-import { PlayersApi } from "../../lib/api"
+import infernals from "../../assets/game/factions/infernals-small-glow.png"
+import vanguard from "../../assets/game/factions/vanguard-small-glow.png"
+import { PlayersApi, Race } from "../../lib/api"
 import { Suspense, createEffect, createResource, createSignal, on, onMount, useTransition } from "solid-js"
 import { Widget } from "../ui/Widget"
 import { Pagination } from "../ui/Pagination"
 import MatchPreview from "./MatchPreview.tsx"
+import { SelectButton, type SelectButtonOption } from "../ui/SelectButton.tsx"
 
 interface Props {
   page?: number
   limit?: number
+  faction?: Race
   playerId: string
 }
 const perPage = 20
 
+const factionOptions: SelectButtonOption[] = [
+  { label: "All Factions", value: "all" },
+  {
+    label: "Infernals",
+    value: Race.INFERNALS,
+    icon: infernals.src,
+  },
+  { label: "Vanguard", value: Race.VANGUARD, icon: vanguard.src },
+]
+
+const getFactionOption = (value: string | undefined) =>
+  factionOptions.find((option) => option.value === value) || factionOptions[0]
```

Comment:
I wonder if I should create a SelectFaction component to factorize this part of the code that is a duplicate from Leaderboard, WDYT?

Reply:
Maybe, I think you will end up prop drilling a lot. Not sure, you're free to try (as a wrapper of Select)

Reply:
Yeah, that is what I was afraid of.

Reply:
Since getFactionOption is used outside of the SelectButton, I am not sure it will bring a lot of value. I guess it is fine like this for now